instances/widgets.treasury-factory.near/widget/pages/lockup/CreateRequest.jsx

```diff
@@ -0,0 +1,422 @@
+const [amountError, setAmountError] = useState(null);
+const [lastProposalId, setLastProposalId] = useState(null);
+
+function formatTimestamp(date) {
+  return new Date(date).getTime() * 1000000;
+}
+
+function toBase64(json) {
+  return Buffer.from(JSON.stringify(json)).toString("base64");
+}
+
+function onSubmitClick() {
+  setTxnCreated(true);
+
+  const deposit = Big(isNaN(amount) ? 0 : parseInt(amount))
+    .mul(Big(10).pow(24))
+    .toFixed();
+  const gas = 270000000000000;
+  const vestingArgs = allowCancellation
+    ? {
+        vesting_schedule: {
+          VestingSchedule: {
+            cliff_timestamp: formatTimestamp(cliffDate || startDate).toString(),
+            end_timestamp: formatTimestamp(endDate).toString(),
+            start_timestamp: formatTimestamp(startDate).toString(),
+          },
+        },
+      }
+    : {
+        lockup_timestamp: formatTimestamp(startDate).toString(),
+        release_duration: (
+          formatTimestamp(endDate) - formatTimestamp(startDate)
+        ).toString(),
+      };
+
+  const calls = [
+    {
+      contractName: treasuryDaoID,
+      methodName: "add_proposal",
+      args: {
+        proposal: {
+          description: `Create lockup for ${receiver}`,
+          kind: {
+            FunctionCall: {
+              receiver_id: "lockup.near",
+              actions: [
+                {
+                  method_name: "create",
+                  args: toBase64(
+                    allowStaking
+                      ? {
+                          lockup_duration: "0",
+                          owner_account_id: receiver,
+                          ...vestingArgs,
+                        }
+                      : {
+                          lockup_duration: "0",
+                          owner_account_id: receiver,
+                          whitelist_account_id: "lockup-no-whitelist.near",
+                          ...vestingArgs,
+                        }
+                  ),
+                  deposit,
+                  gas: "150000000000000",
+                },
+              ],
+            },
+          },
+        },
+      },
+      gas,
+    },
+  ];
+
+  Near.call(calls);
+}
+
+function cleanInputs() {
+  setReceiver("");
+  setAmount("");
+  setStartDate("");
+  setEndDate("");
+  setCliffDate("");
+}
+
+function isValidAmount() {
+  if (
+    isNaN(amount) ||
+    amount === "" ||
+    parseFloat(amount) > balance ||
+    parseFloat(amount) < MINIMUM_AMOUNT
+  )
+    return false;
+
+  return true;
+}
+
+function getLastProposalId() {
+  return Near.asyncView(treasuryDaoID, "get_last_proposal_id").then(
+    (result) => result
+  );
+}
+
+useEffect(() => {
+  getLastProposalId().then((i) => setLastProposalId(i));
+}, []);
+
+// close canvas after proposal is submitted
+useEffect(() => {
+  if (isTxnCreated) {
+    let checkTxnTimeout = null;
+    let errorTimeout = null;
+    const checkForNewProposal = () => {
```

Comment:
we don't need the errorTimeout, we don't use it, please check [this](https://github.com/NEAR-DevHub/neardevhub-treasury-dashboard/blob/staging/instances/widgets.treasury-factory.near/widget/pages/payments/CreatePaymentRequest.jsx#L197)